include/nektar_interface/solver_base/eqnsys_base.hpp

```diff
@@ -39,18 +39,38 @@ class EqnSysBase : public NEKEQNSYS {
       : NEKEQNSYS(session, graph), field_to_index(session->GetVariables()),
         required_fld_names() {
 
-    // If number of particles / number per cell was set in config; construct the
-    // particle system
-    int num_parts_per_cell, num_parts_tot;
-    session->LoadParameter(PartSysBase::NUM_PARTS_TOT_STR, num_parts_tot, -1);
-    session->LoadParameter(PartSysBase::NUM_PARTS_PER_CELL_STR,
-                           num_parts_per_cell, -1);
-    this->particles_enabled = num_parts_tot > 0 || num_parts_per_cell > 0;
-    if (this->particles_enabled) {
-      this->particle_sys = std::make_shared<PARTSYS>(session, graph);
+    /*
+    Particle system type is defined in the same xml document as the Nektar++
+    settings <NEKTAR>
+    ...
+      <PARTICLES>
+        <INFO>
+          <I PROPERTY="PARTTYPE" VALUE="MyParticleSystem"/>
+        </INFO>
+      </PARTICLES>
+    </NEKTAR>
+    */
+
```

Comment:
```suggestion
this->particles_enabled = false;
```
Think you need to initialise particles_enabled before the conditional, otherwise it's undefined for non-particle configs.
When I run SimpleSOLTest.2Drot45 (for instance), which doesn't have a PARTICLES block, the particle setup gets triggered anyway and there's a segfault when trying to report the particle params.

Reply:
I've done this, hopefully tests now pass for you